Update RunWebServerInDockerContainer.rst to use ubuntu and usergroups

Koldstream opened the pull request:
Instructions were outdated and still based on SL7.

Related ticket: [CLDSCRM-535](https://stfc.atlassian.net/browse/CLDSCRM-535?atlOrigin=eyJpIjoiYTQ0NzczNmI1MTVhNGIyNGI4ODY0Y2Y0NzM0NjMyY2IiLCJwIjoiaiJ9)

Changes:
- [X] Updated to be based on Ubuntu focal 20.04, new install instructions for adding Docker apt repositories.
- [X] Advised to add users to Dock group, instead of running docker and containers as root
- [X] Put in a strongly worded message to deter users from using root

Koldstream committed:
Update RunWebServerInDockerContainer.rst to use ubuntu and usergroups

DavidFair commented on source/howto/RunWebServerInDockerContainer.rst:
```diff
@@ -6,42 +6,78 @@ Running an NGinX webserver inside a Docker container
+    "deb [arch="$(dpkg --print-architecture)" signed-by=/etc/apt/keyrings/docker.gpg] https://download.docker.com/linux/ubuntu \
+    "$(. /etc/os-release && echo "$VERSION_CODENAME")" stable" | \
+    sudo tee /etc/apt/sources.list.d/docker.list > /dev/null
+  sudo apt-get update
```
We should just do `apt-get update && apt-get install docker.io`

Koldstream committed:
Update install instructions to use docker.io

Koldstream committed:
Update RunWebServerInDockerContainer.rst

Don't ask users to su - ${USER} because they'll probably just end up doing everything as root. So just exit and ssh back in.
